### PR TITLE
Add info about binary logging

### PR DESCRIPTION
**Pull request by mrquincle**
Created documentation on how to do (binary) logging.

### CODE REVIEW

File: docs/PROTOCOL.md [hunk 1]
```diff
@@ -625,7 +627,7 @@ Type | Name | Length | Description
 uint32 | Min stack end | 4 | Minimal observed stack end pointer since boot. It might take some time before this reflects the actual minimum.
 uint32 | Max heap end | 4 | Maximal observed heap end pointer since boot.
 uint32 | Min free | 4 | Minimal observed free RAM in bytes. It might take some time before this reflects the actual minimum.
-uint32 | Num sbrk fails | 4 | Number of times sbrk failed to hand out space.
+uint32 | Number of sbrk fails | 4 | Number of times sbrk failed to hand out space.
```
Comment (vliedel) [inline docs/PROTOCOL.md:630]: Name can remain shortened version of the description.

File: docs/LOGGING.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Logging
+
+There are two manners with which Crownstone firmware supports logging:
+
+* plain text logging
+* binary logging
+
+The firmware will send logs via UART. The UART device to be used depends on the configuration settings that you set at build or configuration time.
```
Comment (vliedel) [inline docs/LOGGING.md:8]: `The UART device to be used depends on the configuration settings that you set at build or configuration time.` device?

File: docs/LOGGING.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+Some remarks:
+
+* If you attach a Crownstone and you do not know which firmware is on it, you will have a hard time getting the debug information out. Hence, it is a good habit to do regular public commits. For each public commit it is possible to check out that particular commit and then run the given script to be able to get the debug logs.
+* If you attach a Crownstone and you do actually know that it is one with a version on it that is committed, you have to find out a way to get this information out. **Check**: there is no UART command yet to read out the firmware version over UART (see [UART_PROTOCOL](UART_PROTOCOL.md) and [lib doc](https://github.com/crownstone/crownstone-lib-python-uart/blob/master/DOCUMENTATION.md)).
+* If you want to run a development kit or dongle from for example a PI or other type of hardware to do extended logging, you will need to install git, checkout the code base on that device and point the script towards the source files. There is no separate extraction step yet (which would allow you to do this with something like a `logging.db` file which you would generate on a host computer and which will be generally available for our releases).
```
Comment (vliedel) [inline docs/LOGGING.md:46]: Logging is still disabled in releases.

File: docs/LOGGING.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+* If you attach a Crownstone, there's always some output in plain text (so you know UART is operational). However, to actually start sending output, a command has to be sent over UART. It will not start logging automatically. This is done on purpose (we do not want some continuous logging in release).
```
Comment (vliedel) [inline docs/LOGGING.md:47]: There is no command (yet) to enable logging, only to enable UART TX completely. Crownstone dongles have UART TX and RX enabled by default.

File: docs/LOGGING.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+bluenetLogs = BluenetLogs()
+bluenetLogs.setSourceFilesDir(sourceFilesDir)
+```
+
+The `sourceFilesDir` is by default the `source` directory in bluenet.
+
+Some remarks:
+
+* If you attach a Crownstone and you do not know which firmware is on it, you will have a hard time getting the debug information out. Hence, it is a good habit to do regular public commits. For each public commit it is possible to check out that particular commit and then run the given script to be able to get the debug logs.
+* If you attach a Crownstone and you do actually know that it is one with a version on it that is committed, you have to find out a way to get this information out. **Check**: there is no UART command yet to read out the firmware version over UART (see [UART_PROTOCOL](UART_PROTOCOL.md) and [lib doc](https://github.com/crownstone/crownstone-lib-python-uart/blob/master/DOCUMENTATION.md)).
```
Comment (vliedel) [inline docs/LOGGING.md:45]: Correct.

**Review by vliedel — COMMENTED** [review]
Thanks :)

File: docs/PROTOCOL.md (hunk 1 above)
Comment (mrquincle) [reply docs/PROTOCOL.md:630]: Number is abbreviated differently all the time. Num is my least favorite, but then that one should be used consistently.

File: docs/LOGGING.md (hunk 2 above)
Comment (mrquincle) [reply docs/LOGGING.md:8]: Yes, that is a pleonasm indeed. A UART is indeed already a device. It's a receiver-transmitter. :-) I'll reword it.

Comment (vliedel) [reply docs/LOGGING.md]: Ah, now i see what you meant, the device on the pc side.

File: docs/LOGGING.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+* If you want to run a development kit or dongle from for example a PI or other type of hardware to do extended logging, you will need to install git, checkout the code base on that device and point the script towards the source files. There is no separate extraction step yet [1].
+* If you attach a Crownstone, there's always some output in plain text (so you know UART is operational). There is no command to enable logging. It is either enabled (on the Crownstone USB dongle) or not (the other products) [2].
```
Comment (vliedel) [inline docs/LOGGING.md:56]: I guess there's still some confusion:

There's a difference between uart being enabled, and logging being enabled. While crownstone dongles have uart enabled, logs are not even compiled into a release binary (yet).
With uart enabled, you will receive packets like crownstone state, presence updates, etc. This is all the dongles need for the libraries, while logs are just for debugging.

If logs are in the binary, then uart TX still needs to be enabled for them to be written.

**Pull request merged**